fortex/health/processors/coreference_processor.py

```diff
@@ -0,0 +1,228 @@
+        self.set_up(configs)
+
+    def _process(self, input_pack: DataPack):
+        r"""
+        Coreference resolution is done by
+        a spaCy pipeline with `NeuralCoref` added.
+
+        Then we translate the output to `CoreferenceGroup`.
+        """
+
+        def load_module(string):
+            path_str, module_str = string.rsplit(".", 1)
+            mod = importlib.import_module(path_str)
+            return getattr(mod, module_str)
+
+        # Default: Document
+        entry_type = load_module(self.configs.entry_type)
+
+        # Default: MedicalEntityMention
+        mention_type = load_module(self.configs.mention_type)
+
+        for entry_specified in input_pack.get(entry_type=entry_type):
+            result = self.spacy_nlp(entry_specified.text)
+
+            if result._.has_coref:
+                for cluster in result._.coref_clusters:
+
+                    mentions = []
+                    for mention in cluster.mentions:
+                        mention_text = mention.text
+                        mention = mention_type(
+                            input_pack,
+                            mention.start_char + entry_specified.begin,
+                            mention.end_char + entry_specified.begin,
+                        )
+
+                        # TODO: remove assertion?
+                        assert mention.text == mention_text, (
+                            f"The processor extracted mention {mention.text}"
+                            " which is different from the original mention"
+                            f" {mention_text}. The offeset calculation is wrong."
+                        )
+                        mentions.append(mention)
+
+                    group = CoreferenceGroup(input_pack)
+                    group.add_members(mentions)
+
+    @classmethod
+    def default_configs(cls):
+        r"""
+        This defines a basic config structure for `CoreferenceProcessor`.
+
+        Following are the keys for this dictionary:
+        - `entry_type`: Input entry type. You can change the context of
+          coreference resolution by setting this parameter. For example,
+          if you want to do coreference resolution within documents, set
+          it to `"ft.onto.base_ontology.Document"`. If you want to do
+          coreference resolution within sentences, set it to
+          `"ft.onto.base_ontology.Sentence"`.
+          Default: `"ft.onto.base_ontology.Document"`.
+        - `mention_type`: The type of members in `CoreferenceGroup`.
+          It can be set to `"ft.onto.base_ontology.EntityMention"` or
+          its subclasses.
+          Default: `"ftx.medical.clinical_ontology.MedicalEntityMention"`.
+        - `lang`: The SpaCy pipeline to be used. The pipeline does the
+          preprocessing steps for NeuralCoref.
+          Default: `"en_core_web_sm"`.
+        - `model`: the neural net model to be used by NeuralCoref. If set
```

Comment:
Is it the case that the user can pass an actual model object through this `config`? Our config system mainly supports text-based configs, having binary models here may break some functionalities.

Reply:
Yes, it should be a model object. After receiving the object, neuralcoref will load the weights to the object. So I guess we need to remove this argument.